clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy

```diff
@@ -91,4 +99,26 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
 
     task.updateStatus BASE_PHASE, "Successfully destroyed server group $description.serverGroupName."
   }
+
+  void destroyController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, String autoscalerName) {
+    switch(description.kind) {
+      case KubernetesUtil.CONTROLLERS_STATEFULSET_KIND:
+        if (credentials.apiClientAdaptor.getAutoscaler(namespace, autoscalerName)) {
+          task.updateStatus BASE_PHASE, "Destroying autoscaler..."
+          if (!credentials.clientApiAdaptor.deleteAutoscaler(namespace, autoscalerName, null, null, null)) {
+            throw new KubernetesOperationException("Failed to delete associated autoscaler $autoscalerName in $namespace.")
+          }
+        }
+        credentials.apiClientAdaptor.deleteStatefulSetNotCascade(serverGroupName, namespace, null,null, null)
+        break
+      case KubernetesUtil.CONTROLLERS_DAEMONSET_KIND:
+        credentials.apiClientAdaptor.hardDestroyDaemonSet(serverGroupName, namespace, null,null, null)
```

Comment:
Inconsistent spacing around `null`

Reply:
Without delete options or orphan dependents set, will this work?

Reply:
set to `new V1DeleteOption()` it it is null in api call.